src/Config.php

```diff
@@ -52,22 +54,26 @@ public function __construct($hostname, $conf = null, CacheItemPoolInterface $cac
                 'sysvshm extension is required to use this ItemPool');
             }
             $channel_pool_key = intval(base_convert(sha1($channel_pool_key), 16, 10));
-            $shm_id = shm_attach(getmypid(), 200000, 0600);
-            $var1 = @shm_get_var($shm_id, $channel_pool_key);
-            if ($var1) {
-                $gcp_call_invoker = unserialize($var1);
+            $shm_id = $this->get_shmem();
+            if (shm_has_var($shm_id, $channel_pool_key)) {
+                $shm_var = shm_get_var($shm_id, $channel_pool_key);
+                $gcp_call_invoker = unserialize($shm_var);
             } else {
                 $affinity_conf = $this->parseConfObject($conf);
                 $gcp_call_invoker = new GCPCallInvoker($affinity_conf);
             }
             $this->gcp_call_invoker = $gcp_call_invoker;
+            shm_detach($shm_id);
 
             register_shutdown_function(function ($gcp_call_invoker, $channel_pool_key, $shm_id) {
                 // Push the current gcp_channel back into the pool when the script finishes.
+                $shm_id = $this->get_shmem();
```

Comment:
I am not `$shm_id` matters here.
If it doesn't matter, you can remove the `$shm_id` inside `register_shutdown_function` in line 68.
`register_shutdown_function` is called by the end of each script. If `$shm_id` matters, you can call `shm_detach` once inside this function.